runelite-client/src/main/java/net/runelite/client/plugins/timers/TimersPlugin.java

```diff
@@ -669,8 +706,48 @@ public void onNpcDespawned(NpcDespawned npcDespawned)
 		}
 	}
 
-	private void createGameTimer(GameTimer timer)
+	/**
+	 * Adds a FREEZE game timer to the infobox manager to render. Accepts a `replaceExisting` option to
+	 * preserve existing timers of the same type instead of replacing them with the new passed
+	 * timer. Accepts a `checkActiveFreezeTimers` option to preserve existing FREEZE timers of a certain
+	 * freeze spell instead of replacing or adding additional freeze timers of a different freeze spell.
+	 *
+	 * @param timer           				The timer to be added to the infobox manager.
+	 * @param replaceExisting 				Whether to remove existing timer of the same type. `true`
+	 *                                      to replace it, `false` to preserve it if present.
+	 * @param mutuallyExclusiveGameTimers	Mutually exclusive set of timers to be checked. If the timer
+	 *                                      at hand is in the set. It will not be created.
+	 */
+
+	private void createGameTimer(final GameTimer timer, final boolean replaceExisting, final Set<GameTimer> mutuallyExclusiveGameTimers)
+	{
+		if (!mutuallyExclusiveGameTimers.isEmpty()
+			&& infoBoxManager.getInfoBoxes().stream().anyMatch(t -> t instanceof TimerTimer && mutuallyExclusiveGameTimers.contains(((TimerTimer) t).getTimer())))
```

Comment:
I would rather you to not use the .getInfoboxes() like this anywhere in this class. Instead add method .has(Function<InfoBox, Timer>) to infoboxManager and use that

Reply:
done!